File: URI/Lua/VariasNotasComValidacao.lua
media = 0
qte = 0
novoC = 1
while novoC == 1 do
    nota = io.read("*number")
    if 0 <= nota and nota <= 10 then
        media = media + nota
        qte = qte + 1
    else
        print("nota invalida")
    end

    if qte == 2 then
        print(string.format( "media = %.2f",media/2 ))
        media = 0
        qte = 0
        ::LABEL1::
        print("novo calculo (1-sim 2-nao)")
        novoC = io.read("*number")
        if novoC ~= 1 and novoC ~= 2 then
            goto LABEL1
        end
    end
end

File: URI/Python/VariasNotasComValidacao.py
media = 0
qte = 0
novoCalc = 1
while novoCalc == 1:
    nota = float(input())
    if 0 <= nota <= 10:
        media += nota
        qte += 1
    else:
        print("nota invalida")

    if qte == 2:
        print("media = {:.2f}".format(media/2))
        media = 0
        qte = 0
        novoCalc = -1
        while novoCalc != 1 and novoCalc != 2:
            print("novo calculo (1-sim 2-nao)") 
            novoCalc = int(input())   
